### PR TITLE
fix HarmonyAcceptImportDependency

### LINKED ISSUE #14974: HMR breaks async module graph + dynamic import (Top level await or Async WebAssembly)
# Bug report

**What is the current behavior?**

Webpack generates invalid code:

> Uncaught (in promise) ReferenceError: _defer_js__WEBPACK_IMPORTED_MODULE_1__ is not defined

**If the current behavior is a bug, please provide the steps to reproduce.**

Reproduction: https://github.com/Jack-Works-forks/webpack-hmr-async-module-bug

```sh
npx webpack serve --mode development --hot
```

**What is the expected behavior?**

Works normally

**Other relevant information:**
webpack version:  5.65.0
Node.js version: 17.2.0

### PR DESCRIPTION
**Pull request by vankop**
**What kind of change does this PR introduce?**
fixes #14974 

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
nothing

### CODE REVIEW

File: lib/dependencies/HarmonyAcceptImportDependency.js [hunk 1]
```diff
@@ -30,6 +30,14 @@ makeSerializable(
...
 	HarmonyImportDependency.Template
-) {};
+) {
+	/**
+	 * @param {Dependency} dependency the dependency for which the template should be applied
+	 * @param {ReplaceSource} source the current replace source which can be modified
+	 * @param {DependencyTemplateContext} templateContext the context object
+	 * @returns {void}
+	 */
+	apply(dependency, source, templateContext) {}
```
Comment (vankop) [inline lib/dependencies/HarmonyAcceptImportDependency.js:40]: could be used NullDependency template, but ts does not like this..

**Comment by sokra** [comment]
Something doesn't work on node 10. Maybe you have to filter that version as it is missing `async` support?

**Comment by vankop** [comment]
test behaves weird on Node.js 10. It looks like [`_require`](https://github.com/webpack/webpack/blob/main/test/ConfigTestCases.template.js#L459) finish before actual finish.. 🤯

**Review by sokra — APPROVED** [review]

**Comment by sokra** [comment]
Thansk

**Pull request merged**